cmd/codeqlExecuteScan.go

```diff
@@ -185,13 +187,50 @@ func uploadResults(config *codeqlExecuteScanOptions, repoInfo RepoInfo, token st
 
 	//if no git pramas are passed(commitId, reference, serverUrl, repository), then codeql tries to auto populate it based on git information of the checkout repository.
 	//It also depends on the orchestrator. Some orchestrator keep git information and some not.
+
+	var buffer bytes.Buffer
+	utils.Stdout(&buffer)
 	err := execute(utils, cmd, GeneralConfig.Verbose)
 	if err != nil {
 		log.Entry().Error("failed to upload sarif results")
-		return err
+		return "", err
 	}
+	utils.Stdout(log.Writer())
+
+	url := buffer.String()
+	return strings.TrimSpace(url), nil
+}
 
-	return nil
+func waitSarifUploaded(codeqlSarifUploader codeql.CodeqlSarifUploader) error {
+	var sarifUploadComplete = "complete"
+	var sarifUploadFailed = "failed"
+	var maxRetries = 5
+	var retryCount = 0
+	for {
+		sarifStatus, err := codeqlSarifUploader.GetSarifStatus()
+		if err != nil {
+			if retryCount < maxRetries {
+				retryCount++
+				log.Entry().Errorf("error while checking sarif status, retrying in 10 seconds... (retry %d/%d)", retryCount, maxRetries)
+				time.Sleep(time.Second * 10)
+				continue
+			} else {
+				return err
+			}
+		}
+		if len(sarifStatus.Errors) > 0 {
```

Comment:
the if does not seem necessary to me - it does not add anything to the for loop.

the other question is if this code gets executed at all. in the error case, the control flow always seems to go elsewhere via a return or a continue above. or am i missing something?

Reply:
this code is executed when the server correctly responds with processing status "failed", and it will not be executed if there was an error while reading the server's response